src/main/java/com/piseth/java/school/schoolManagement/service/impl/MonthlyScoreServiceImpl.java

```diff
@@ -116,4 +115,28 @@ private RankDTO toRankDTO(Student student, List<MonthlyScore> mapStudent) {
 		return rankDTO;
 	}
 
+	@Override
+	public Map<String,Double> getListScoreByStudentIDYearMonth(Map<String, String> params) {
+			MonthlyScoreFilter monthlyScoreFilter = new  MonthlyScoreFilter();
+			
+			if(params.containsKey(MonthlyScorePropertyFilter.STUDENTID)) {
+				monthlyScoreFilter.setAStudentId(params.get(MonthlyScorePropertyFilter.STUDENTID).isBlank() ? 0 : Long.parseLong(params.get("studentId")));
+			}
+			if(params.containsKey(YEAR)) {
+				monthlyScoreFilter.setYear(params.get(YEAR).isBlank() ? 0 : Short.parseShort(params.get(YEAR)));
+			}
+			if(params.containsKey(MONTH)) {
+				monthlyScoreFilter.setMonth(params.get(MONTH).isBlank() ? 0 : Short.parseShort(params.get(MONTH)));
+			}
+			
+			MonthlyScoreSpec monthlyScoreSpec = new MonthlyScoreSpec(monthlyScoreFilter);
+			return toListSubjectScore(monthlyScoreRepository.findAll(monthlyScoreSpec));
+	}
+	
+	private Map<String,Double> toListSubjectScore(List<MonthlyScore> monthlyScores) {
+			Map<String,Double> listSubjectscore = new HashMap<>();
+			monthlyScores.forEach(l->listSubjectscore.put(l.getSubject().getName(),l.getScore()));
```

Comment:
To convert from list to map, you can use stream().collect(Collectors.toMap(.......)).
Please google how to do it.